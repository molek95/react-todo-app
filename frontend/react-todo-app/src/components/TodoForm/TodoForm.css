.todo-form {
    margin-bottom: 32px;
}

.todo-input {
    padding: 14px 32px;
    border-radius: 5px 0 0 5px;
    border: 2px solid #5d0cff;
    outline: none;
    width: 300px;
    background: transparent;
    color: #fff;
}

.todo-input::placeholder {
    color: #e2e2e2;
}

.todo-button {
    padding: 16px;
    border: none;
    border-radius: 0 5px 5px 0;
    outline: none;
    cursor: pointer;
    background: linear-gradient(
        90deg,
        rgba(93, 12, 255, 1) 0%
        rgba(155, 0, 250, 1) 100%
    );
    text-transform: capitalize;
}

.todo-input.edit {
    border: 2px solid #149fff;
}

.todo-button.edit {
    background: linear-gradient(
        90deg,
        rgba(20, 159, 255, 1) 0%
        rgba(17, 122, 255, 1) 100%
    );
    padding: 10px;
}