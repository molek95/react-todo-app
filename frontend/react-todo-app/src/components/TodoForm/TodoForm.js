import React, { useState, useEffect, useRef } from 'react'
import './TodoForm.css';

function TodoForm(props) {
    const [input, setInput] = useState('')
    const inputRef = useRef(null)

    useEffect(() => {
        inputRef.current.focus()
    })

    const submitTodo = e => {
        e.preventDefault();
        
        props.onSubmit({
            id: Math.floor(Math.random() * 1000),
            text: input
        });

        setInput('')
    }

    const handleChange = e => {
        setInput(e.target.value);
    }

    return (
        <form className="todo-form" onSubmit={submitTodo}>
            <input 
                type="text" 
                placeholder="Add a todo" 
                value={input} 
                name="text" 
                className="todo-input"
                onChange={handleChange}
                ref={inputRef}/>
            <button className="todo-button">Add todo</button>
        </form>
    )
}

export default TodoForm
