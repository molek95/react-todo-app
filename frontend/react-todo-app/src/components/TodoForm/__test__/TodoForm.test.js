import React from 'react';
import { shallow, mount } from 'enzyme';
import TodoForm from './../TodoForm';

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

Enzyme.configure({ adapter: new Adapter() });

describe('<TodoForm /> with no props', () => {
    const todoForm = shallow(<TodoForm />);

    it('should match the snapshot', () => {
        expect(todoForm.html()).toMatchSnapshot();
    });

    it('should have an input field', () => {
        expect(todoForm.find('input[type="text"]').length).toEqual(1);
    });

    it('should have an add todo button', () => {
        expect(todoForm.find('button').length).toEqual(1);
    });

    it('should trigger state change on submit', () => {
        const submitTodo = jest.fn();
        const form = mount(<TodoForm onSubmit={submitTodo} />)
        const handleClick = jest.spyOn(React, "useState")

        handleClick.mockImplementation(todo => [todo, submitTodo]);
        form.find('button').simulate('click');
        expect(submitTodo).toBeTruthy();
    })
})
