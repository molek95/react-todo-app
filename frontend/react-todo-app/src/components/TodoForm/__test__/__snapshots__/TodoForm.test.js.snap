// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoForm /> with no props should match the snapshot 1`] = `"<form class=\\"todo-form\\"><input type=\\"text\\" placeholder=\\"Add a todo\\" value=\\"\\" name=\\"text\\" class=\\"todo-input\\"/><button class=\\"todo-button\\">Add todo</button></form>"`;
